### PR TITLE
[RFC] Change return type to bool on undo.c functions

### PR DESCRIPTION
**Pull request by war1025**

### CODE REVIEW

File: src/nvim/undo.c [hunk 1]
```diff
@@ -757,6 +757,10 @@ static void u_free_uhp(u_header_T *uhp)
 }
 
 /// Writes the header.
```
Comment (bfredl) [inline src/nvim/undo.c]: maybe "the undofile header" (to distinguish from "undo header" below)

File: src/nvim/undo.c [hunk 2]
```diff
@@ -810,6 +814,12 @@ static bool serialize_header(bufinfo_T *bi, char_u *hash)
   return true;
 }
 
+/// Writes the undo header.
```
Comment (bfredl) [inline src/nvim/undo.c]: maybe "an undo header" (as we except many undo headers, unlike _the_ undofile header)

File: src/nvim/undo.c [hunk 3]
```diff
@@ -1565,6 +1582,10 @@ static bool undo_write(bufinfo_T *bi, uint8_t *ptr, size_t len)
 ///
 /// Must match with undo_read_?c() functions.
 ///
+/// @param bi  The buffer info
+/// @param nr  The number to write
+/// @param len The number of bytes to use
```
Comment (bfredl) [inline src/nvim/undo.c]: number of bytes to write?

File: src/nvim/undo.c [hunk 4]
```diff
@@ -2839,19 +2864,26 @@ static char_u *u_save_line(linenr_T lnum)
...
     (buf->b_changed || file_ff_differs(buf, true));
 }
 
-int curbufIsChanged(void)
+/// Check if the 'modified' flag is set, or 'ff' has changed (only need to
+/// check the first character, because it can only be "dos", "unix" or "mac").
+/// "nofile" and "scratch" type buffers are considered to always be unchanged.
+///
+/// @return true if the current buffer has changed
+bool curbufIsChanged(void)
```
Comment (bfredl) [inline src/nvim/undo.c]: (not in this PR but later on I suppose we could replace all calls to this with `bufIsChanged(curbuf)` )

**Comment by bfredl** [comment]
some minor suggestions otherwise LGTM

File: src/nvim/undo.c [hunk 5]
```diff
@@ -284,32 +284,32 @@ int u_savedel(linenr_T lnum, long nlines)
       nlines == curbuf->b_ml.ml_line_count ? 2 : lnum, FALSE);
 }
 
-/*
- * Return TRUE when undo is allowed.  Otherwise give an error message and
- * return FALSE.
- */
-int undo_allowed(void)
+/// Return true when undo is allowed.  Otherwise give an error message and
```
Comment (Pyrohh) [inline src/nvim/undo.c]: `give -> `print` ?

**Comment by Pyrohh** [comment]
Besides the nit, this LGTM.  Since @bfredl said so as well, I'll merge this soon.

**Comment by war1025** [comment]
@Pyrohh Just noticed your comment. Fixed your nitpick.

**Comment by Pyrohh** [comment]
@war1025 Committed, thanks!

**Pull request closed without merging**